test/gtest/ucs/test_memtype_cache.cc

```diff
@@ -81,8 +81,14 @@ class test_memtype_cache : public ucs::test_with_param<ucs_memory_type_t> {
         test_lookup_found(ptr, 0, expected_type);
     }
 
-    void test_region_found(const mem_buffer &b) const {
-        test_ptr_found(b.ptr(), b.size(), b.mem_type());
+    void test_region_found(const mem_buffer &b, bool cache_updated) const {
+        ucs_memory_type_t mem_type = b.mem_type();
+        if (!cache_updated) {
+            mem_type = ((b.mem_type() == UCS_MEMORY_TYPE_CUDA) ||
+                        (b.mem_type() == UCS_MEMORY_TYPE_CUDA_MANAGED)) ?
+                        UCS_MEMORY_TYPE_LAST : b.mem_type();
```

Comment:
why need special case for cuda?
this test should now pass if either b.mem_type **or** TYPE_UNKNOWN are found in the cache